index.js

```diff
@@ -0,0 +1,251 @@
+
+            return;
+        }
+
+        if (typeof this.scms[scmContext] === 'object') {
+            console.warn(`Duplicate scm config for ${scmContext}`);
+        }
+
+        this.scms[scmContext] = scmPlugin;
+    }
+
+    /**
+     * choose webhook scm module
+     * @method chooseWebhookScm
+     * @param  {Object}     headers          The request headers associated with the webhook payload
+     * @param  {Object}     payload          The webhook payload received from the SCM service
+     * @return {Promise}                     scm object
+     */
+    chooseWebhookScm(headers, payload) {
+        return new Promise((resolve, reject) => {
+            // choose a webhook scm module, or null if there is no suitable one
+            async.detectSeries(this.scms, (scm, cb) => {
+                scm.canHandleWebhook(headers, payload)
+                .then((result) => {
+                    cb(result === false ? null : scm);
+                }).catch(() => {
+                    cb(null);
+                });
+            }, (ret) => {
+                if (ret == null) {
+                    reject('there is no suitable webhook module');
+                }
+
+                resolve(ret);
+            });
+        });
+    }
+
+    /**
+     * choose scm module
+     * @method chooseScm
+     * @param  {Object}     config              Configuration
+     * @param  {String}     config.scmContext   Name of scm context
+     * @return {Promise}                        scm object
+     */
+    chooseScm(config) {
+        return new Promise((resolve, reject) => {
+            if (config.scmContext == null || config.scmContext.length === 0) {
+                reject('Not implemented');
+            }
+
+            const scm = this.scms[config.scmContext];
+
+            if (typeof scm !== 'object') {
+                reject('Not implemented');
+            }
+
+            resolve(scm);
+        });
+    }
+
+    /**
+     * Process by all scm module
+     * @method allScm
+     * @param  {Function}   callback            {Map} fn(scm)
+     * @return {Promise}                        combined callback results
+     */
+    allScm(callback) {
+        return new Promise((resolve, reject) => {
+            async.mapSeries(Object.keys(this.scms), (key, cb) => {
+                const scm = this.scms[key];
+
+                callback(scm)
+                .then((ret) => {
+                    cb(null, ret);
+                }, err => cb(err));
+            }, (err, results) => {
+                let map = {};
+
+                if (err) {
+                    reject(err);
+                }
+
+                results.forEach((result) => {
+                    map = Object.assign(map, result);
+                });
+
+                resolve(map);
+            });
+        });
+    }
+
+    _addWebhook(config) {
+        return this.chooseScm(config).then(scm => scm.addWebhook(config));
+    }
+
+    _parseUrl(config) {
```

Comment:
`scmContext` changed from required to optional.
https://github.com/screwdriver-cd/data-schema/pull/155/commits/ad694c46cc818c7019eac8384a7e0cf30f3e30c9

I think that most problems can be solved by this, whether there will be further need to create dedicated data-schema for `scm-router`, How do you think?